homeassistant/config_entries.py

```diff
@@ -527,10 +527,22 @@ async def async_setup(
         integration: loader.Integration | None = None,
     ) -> None:
         """Set up an entry."""
-        current_entry.set(self)
         if self.source == SOURCE_IGNORE or self.disabled_by:
             return
 
+        current_entry.set(self)
+        try:
+            await self.__async_setup_with_context(hass, integration=integration)
+        finally:
+            current_entry.set(None)
+
+    async def __async_setup_with_context(
+        self,
+        hass: HomeAssistant,
+        *,
```

Comment:
I think we can drop the kwargs for the internal method since the outer caller will always pass them

Reply:
> I think we can drop the kwargs for the internal method and make them all required since the outer caller will always pass them

Done